core/ui/src/main/kotlin/com/merxury/blocker/core/ui/rule/GeneralRulesList.kt

```diff
@@ -28,24 +28,32 @@ import androidx.compose.foundation.layout.windowInsetsBottomHeight
 import androidx.compose.foundation.lazy.LazyColumn
 import androidx.compose.foundation.lazy.items
 import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Surface
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.PreviewParameter
 import androidx.compose.ui.unit.dp
 import com.merxury.blocker.core.designsystem.component.ThemePreviews
 import com.merxury.blocker.core.designsystem.component.scrollbar.FastScrollbar
 import com.merxury.blocker.core.designsystem.component.scrollbar.rememberDraggableScroller
 import com.merxury.blocker.core.designsystem.component.scrollbar.scrollbarState
 import com.merxury.blocker.core.designsystem.theme.BlockerTheme
 import com.merxury.blocker.core.model.data.GeneralRule
+import com.merxury.blocker.core.ui.R
+import com.merxury.blocker.core.ui.previewparameter.RuleListPreviewParameterProvider
 
 @Composable
 fun GeneralRulesList(
     rules: List<GeneralRule>,
     modifier: Modifier = Modifier,
     onClick: (String) -> Unit = {},
 ) {
+    val matchedRuleList = rules.filter { it.matchedAppCount > 0 }
```

Comment:
Composable functions can run very frequently, as often as for every frame of an animation. For this reason, you should do as little calculation in the body of your composable as you can.  
https://developer.android.com/develop/ui/compose/performance/bestpractices